sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/data_transfer.py

```diff
@@ -47,25 +47,34 @@
 
 
 def _build_source_sink(io_dict: Union[Dict, Database, FileSystem]):
-    if not io_dict:
+    if io_dict is None:
         return io_dict
-    io_dict = io_dict or {}
     if isinstance(io_dict, (Database, FileSystem)):
         component_io = io_dict
     else:
-        data_type = io_dict.get("type", None)
-        if data_type == ExternalDataType.DATABASE:
-            component_io = Database(**io_dict)
-        elif data_type == ExternalDataType.FILE_SYSTEM:
-            component_io = FileSystem(**io_dict)
+        if isinstance(io_dict, dict):
+            data_type = io_dict.get("type", None)
+            if data_type == ExternalDataType.DATABASE:
+                component_io = Database(**io_dict)
+            elif data_type == ExternalDataType.FILE_SYSTEM:
+                component_io = FileSystem(**io_dict)
+            else:
+                msg = "Type in source or sink only support {} and {}, currently got {}."
+                raise ValidationException(
+                    message=msg.format(ExternalDataType.DATABASE, ExternalDataType.FILE_SYSTEM, data_type),
+                    no_personal_data_message=msg.format(
+                        ExternalDataType.DATABASE, ExternalDataType.FILE_SYSTEM, "data_type"
+                    ),
+                    target=ErrorTarget.DATA_TRANSFER_JOB,
+                    error_category=ErrorCategory.USER_ERROR,
+                    error_type=ValidationErrorType.INVALID_VALUE,
+                )
         else:
-            msg = "Source or sink only support type {} and {}, currently got {}."
+            msg = "Source or sink only support dict, Database and FileSystem"
```

Comment:
only support dict, Database and FileSystem?

Reply:
Yes, for sink and source, we only allow dict, Database and FileSystem. 
If pass dict, we will convert to Database or FileSystem.

Any better ideas for the error message?

Reply:
mmm.. let's keep it this way.